### PR TITLE
Storage get count

### PR DESCRIPTION
**Pull request by ChristianAgha**

### CODE REVIEW

File: api/v1/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!flask/bin/python
+"""
+init defines app_views
+"""
+from flask import Blueprint
+from api.v1.views.index import *
```
Comment (johncoleman83) [inline api/v1/__init__.py]: it might be cleaner if you alphabetize this, just a suggestion?

File: api/v1/app.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from models import storage
+from api.v1.views import app_views, states
+
+app = Flask(__name__)
+app.register_blueprint(app_views)
+
+
+@app.teardown_appcontext
+def app_teardown(err):
+    return storage.close()
```
Comment (johncoleman83) [inline api/v1/app.py]: please add comments for all functions

File: api/v1/views/__init__.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/python3
+"""
+init.py for views folder
+"""
+from flask import Blueprint, render_template
+
+app_views = Blueprint('app_Views', __name__, url_prefix='/api/v1')
+from api.v1.views.index import *
+from api.v1.views import state
```
Comment (johncoleman83) [inline api/v1/views/__init__.py]: isn't this supposed to be a wildcard import?

File: api/v1/views/index.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+"""
+from flask import Blueprint, render_template, make_response, jsonify
+from api.v1.views import app_views
+from models import city, place, review, state, user
+from models import storage, base_model, amenity
+
+
+@app_views.route('/status')
+def index():
+    return make_response(jsonify({'status': 'OK'}))
```
Comment (johncoleman83) [inline api/v1/views/index.py]: please add comments for all functions

File: models/engine/db_storage.py [hunk 5]
```diff
@@ -78,3 +80,22 @@ def close(self):
             calls remove() on private session attribute (self.session)
         """
         self.__session.remove()
+
+    def get(self, cls, id):
+        """
+        Returns the object based on the class name and its ID
+        """
+        if cls:
```
Comment (johncoleman83) [inline models/engine/db_storage.py]: What if `id` is None?

File: models/engine/db_storage.py [hunk 6]
```diff
@@ -78,3 +80,22 @@ def close(self):
...
+            obj_class = self.__session.query(self.CNC.get(cls)).all()
```
Comment (johncoleman83) [inline models/engine/db_storage.py]: No need to query here, you can simply use the `.all()` function on `self`...  Also, remember `.all()` is setup to take `cls` as an argument, if you use that argument you will not have to rewrite so much code.

File: models/engine/db_storage.py [hunk 7]
```diff
@@ -78,3 +80,22 @@ def close(self):
...
+                if item.__class__.__name__ == cls:
+                    if item.id == id:
+                        return item
+        return None
+
+    def count(self, cls=None):
+        """
+        Returns the number of objects in storage matching the given class name.
+        """
+        all_objects = self.all(cls)
```
Comment (johncoleman83) [inline models/engine/db_storage.py]: You need to properly handle when `cls = None`

File: models/engine/file_storage.py [hunk 8]
```diff
@@ -30,6 +30,12 @@ class FileStorage:
 
     def all(self, cls=None):
         """returns private attribute: __objects"""
+        if cls:
```
Comment (johncoleman83) [inline models/engine/file_storage.py]: Good update here!

File: models/engine/file_storage.py [hunk 9]
```diff
@@ -59,15 +65,34 @@ def reload(self):
...
+        Returns the object based on the class name and its ID
+        """
+        if cls and id:
+            objs = self.all(cls)
+            for class_id, obj in objs.items():
+                if class_id.split(".")[1] == id:
+                    return (objs[class_id])
+            return None
+
+    def count(self, cls=None):
```
Comment (johncoleman83) [inline models/engine/file_storage.py]: Same comments as other Storage engine apply here

**Review by johncoleman83 — CHANGES_REQUESTED** [review]

**Comment by johncoleman83** [comment]
Great updates, please review my comments and respond with new commits or `git squash` or `git ammend` the previous comments, then I'll accept your PR, also please add the proper file `code_review.txt` with your github username in it, since I reviewed you.

**Review by johncoleman83 — APPROVED** [review]

File: api/v1/app.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/python
+"""
+register blueprint for flask app
+"""
+from flask import Flask, jsonify, Blueprint, make_response
+from models import storage
+from api.v1.views import app_views, states
```
Comment (clturner) [inline api/v1/app.py]: Please alphabetize the imports

File: api/v1/views/index.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python3
+"""
+create routes for /status
+"""
+from flask import Blueprint, render_template, make_response, jsonify
```
Comment (clturner) [inline api/v1/views/index.py]: alphabetize imports

File: api/v1/views/states.py [hunk 12]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python
+"""
+Handles all default RestFul API actions for State class
+"""
+from flask import Flask, jsonify, Blueprint, make_response, render_template
```
Comment (clturner) [inline api/v1/views/states.py]: alphabetize imports

File: api/v1/views/states.py [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    for state in all_st:
+        if (state.get("id") == state_id):
+            return jsonify(state)
+    return not_found(404)
+
+
+@app_views.route('/states/<state_id>', methods=['DELETE'], strict_slashes=False)
+def delete_State(state_id=None):
+    """delete state using ID"""
+
```
Comment (clturner) [inline api/v1/views/states.py]: add code for delete function to delete state matching id

File: api/v1/views/states.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+from flask import Flask, jsonify, Blueprint, make_response, render_template
+from api.v1.views import app_views
+from models import state, storage
+
+
+@app_views.route('/states', methods=['GET'], strict_slashes=False)
+def get_States():
+    """get all states"""
+    all_st = [st.to_json() for st in storage.all("State").values()]
+    return jsonify(all_st)
```
Comment (clturner) [inline api/v1/views/states.py:14]: change to:    return jsonify([st.to_json() for st in storage.all("State").values()]) to make one line.

File: models/engine/file_storage.py [hunk 15]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3#!/usr/bin/python3
```
Comment (papamuziko) [inline models/engine/file_storage.py:1]: Heum...

**Pull request merged**